scheduler/scheduler-server/src/test/java/functionaltests/job/error/TestTaskRestartOnNodeFailure.java

```diff
@@ -151,6 +152,48 @@ private void testTaskKillNode(FileLock fileLock, boolean waitBeforeKill) throws
         newNode.kill();
     }
 
+    private void testTaskKillTaskLauncher(FileLock fileLock) throws Exception {
+        Path fileLockPath = fileLock.lock();
+        TestNode nodeToDeploy = startNode();
+
+        log("Submit job");
+        final JobId jobId = schedulerHelper.submitJob(createJob(fileLockPath.toString()));
+
+        log("Wait when node becomes busy");
+        RMNodeEvent event;
+        do {
+            event = schedulerHelper.waitForAnyNodeEvent(RMEventType.NODE_STATE_CHANGED, TIMEOUT);
+        } while (!event.getNodeState().equals(NodeState.BUSY));
+
+        log("Wait when task starts");
+        schedulerHelper.waitForEventTaskRunning(jobId, "Test task");
+
+        /*
+         * Wait some time to make sure the TaskLauncher is deployed
+         */
+
+        log("Wait some time");
+        Thread.sleep(5000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)